ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ExecutionRequestsDataCodec.java

```diff
@@ -34,44 +32,47 @@ public ExecutionRequestsDataCodec(final ExecutionRequestsSchema executionRequest
   /**
    * Decodes the execution requests received from the EL.
    *
-   * @param executionRequestData list of encoded execution requests from the EL
+   * @param executionRequests list of encoded execution requests from the EL
    * @return an ExecutionRequests object with the requests
    */
-  public ExecutionRequests decode(final List<Bytes> executionRequestData) {
-    if (executionRequestData.size() != EXPECTED_REQUEST_DATA_ELEMENTS) {
-      throw new IllegalArgumentException(
-          "Invalid number of execution request data elements: expected "
-              + EXPECTED_REQUEST_DATA_ELEMENTS
-              + ", received "
-              + executionRequestData.size());
-    }
-
+  public ExecutionRequests decode(final List<Bytes> executionRequests) {
     final ExecutionRequestsBuilder executionRequestsBuilder =
         new ExecutionRequestsBuilderElectra(executionRequestsSchema);
 
-    for (int index = 0; index < executionRequestData.size(); index++) {
-      // The request type is implicitly defined as the index of the element in executionRequestData
-      switch ((byte) index) {
+    byte previousRequestType = -1;
+    for (final Bytes request : executionRequests) {
+      if (request.isEmpty()) {
+        throw new IllegalArgumentException("Execution request data must not be empty");
+      }
+      final byte requestType = request.get(0);
+      if (requestType <= previousRequestType) {
+        throw new IllegalArgumentException(
+            "Execution requests are not in strictly ascending order");
+      }
+      final Bytes requestData = request.slice(1);
```

Comment:
This will allow requests with empty request _data_.

For example, this will pass when it should throw an exception.

```java
  @Test
  public void decodeExecutionRequestsDataWithEmptyRequestData() {
    final ExecutionRequests executionRequests =
        codec.decode(List.of(Bytes.of(DepositRequest.REQUEST_TYPE)));
    assertThat(executionRequests.getDeposits()).isEmpty();
  }
```

The spec is a little unclear about this distinction.

* The first assert checks that the request type (`request[0:1]`) is valid.
  * If the `request` is empty, the type will be `b""` and this will throw.
* The second assert ensures the request _data_ (`request[1:]`) is not empty.

```python
request_type, request_data = request[0:1], request[1:]

# Check that the request type is valid
assert request_type in request_types
# Check that the request data is not empty
assert len(request_data) != 0
```

Reply:
Thanks @jtraglia !

Reply:
Fixed on https://github.com/Consensys/teku/pull/8880